app/controllers/attachments_controller.rb

```diff
@@ -19,7 +19,8 @@ def new
 
   action_auth_level :create, :instructor
   def create
-    @attachment = Attachment.new(course_id: @course.id, assessment_id: params[:assessment_id])
+    assessment = Assessment.find_by_name(params[:assessment_id])
```

Comment:
This could be problematic, since assessment names aren't unique.  You need to scope this under the current course (@course.assessments.find_by_name)